lib/basedriver/driver.js

```diff
@@ -51,6 +56,32 @@ class BaseDriver extends MobileJsonWireProtocol {
 
     // allow subclasses to have internal drivers
     this.managedDrivers = [];
+
+    // store event timings
+    this._eventHistory = {
+      commands: [] // commands get a special place
+    };
+  }
+
+  /*
+   * make eventHistory a property and return a cloned object so a consumer can't
+   * inadvertently change data outside of logEvent
+   */
+  get eventHistory () {
+    return _.cloneDeep(this._eventHistory);
+  }
+
+  /*
+   * API method for driver developers to log timings for important events
+   */
+  logEvent (eventName) {
+    if (eventName === "commands") {
+      throw new Error("Cannot log commands directly");
+    }
+    if (typeof eventName !== "string") {
+      throw new Error(`Invalid eventName ${eventName}`);
+    }
+    this._eventHistory[eventName] = Date.now();
```

Comment:
This is a good question. I was thinking we would only want one event of each type, but you raise a good point that an event might reasonably happen more than once and we'd be interested in tracking that. what about having the value of an event key be an array of timestamps rather than a single timestamp?